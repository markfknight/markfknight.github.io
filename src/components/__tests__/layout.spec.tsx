import React, { PropsWithChildren } from 'react';
import { cleanup, render } from '@testing-library/react';

import Layout, { LayoutProps } from '../layout';

afterEach(cleanup);

describe(`Layout snapshot`, () => {
  it(`renders correctly when root pathname`, () => {
    const props = {
      title: `test`,
      location: {
        pathname: `/`,
      },
    } as PropsWithChildren<LayoutProps>;

    const tree = render(
      <Layout {...props}>
        <p>Test</p>
      </Layout>
    );
    expect(tree).toMatchSnapshot();
  });

  it(`renders correctly when not root pathname`, () => {
    const props = {
      title: `test`,
      location: {
        pathname: ``,
      },
    } as PropsWithChildren<LayoutProps>;

    const tree = render(
      <Layout {...props}>
        <p>Test</p>
      </Layout>
    );
    expect(tree).toMatchSnapshot();
  });
});
