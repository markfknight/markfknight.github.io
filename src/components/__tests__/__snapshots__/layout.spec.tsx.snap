// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout snapshot renders correctly when not root pathname 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container max-w-3xl px4 py10"
      >
        <header
          class="py-10"
        >
          <h3>
            <a
              href="/"
            >
              test
            </a>
          </h3>
        </header>
        <main>
          <p>
            Test
          </p>
        </main>
        <footer
          class="py-10 text-center"
        >
          © 
          2021
          , Built with
           
          <a
            href="https://www.gatsbyjs.org"
          >
            Gatsby
          </a>
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container max-w-3xl px4 py10"
    >
      <header
        class="py-10"
      >
        <h3>
          <a
            href="/"
          >
            test
          </a>
        </h3>
      </header>
      <main>
        <p>
          Test
        </p>
      </main>
      <footer
        class="py-10 text-center"
      >
        © 
        2021
        , Built with
         
        <a
          href="https://www.gatsbyjs.org"
        >
          Gatsby
        </a>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Layout snapshot renders correctly when root pathname 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container max-w-3xl px4 py10"
      >
        <header
          class="py-10"
        >
          <h1>
            <a
              href="/"
            >
              test
            </a>
          </h1>
        </header>
        <main>
          <p>
            Test
          </p>
        </main>
        <footer
          class="py-10 text-center"
        >
          © 
          2021
          , Built with
           
          <a
            href="https://www.gatsbyjs.org"
          >
            Gatsby
          </a>
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container max-w-3xl px4 py10"
    >
      <header
        class="py-10"
      >
        <h1>
          <a
            href="/"
          >
            test
          </a>
        </h1>
      </header>
      <main>
        <p>
          Test
        </p>
      </main>
      <footer
        class="py-10 text-center"
      >
        © 
        2021
        , Built with
         
        <a
          href="https://www.gatsbyjs.org"
        >
          Gatsby
        </a>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
